src/util/TaskQueue.h

```diff
@@ -129,13 +129,15 @@ class TaskQueue {
   // value(void) as well as for functions with a return value. See the
   // applications above.
   template <typename F>
-  decltype(auto) executeAndUpdateTimer(F&& f, std::atomic<size_t>& duration) {
+  decltype(auto) executeAndUpdateTimer(
+      F&& f, std::atomic<std::chrono::milliseconds::rep>& duration) {
     if constexpr (TrackTimes) {
       struct T {
         ad_utility::Timer _t{ad_utility::Timer::Started};
-        std::atomic<size_t>& _target;
-        T(std::atomic<size_t>& target) : _target(target) {}
-        ~T() { _target += _t.msecs(); }
+        std::atomic<std::chrono::milliseconds::rep>& _target;
+        T(std::atomic<std::chrono::milliseconds::rep>& target)
+            : _target(target) {}
```

Comment:
Add a typedef for `milliseconds::rep` into this class and the `BatchedPipeline` header, then the code gets shorter.